src/modules/configs/config.py

```diff
@@ -81,3 +81,120 @@ class OperatorConf(Conf):
 
     def __init__(self, conf_path: Union[str, bool]):
         super().__init__(conf_path, "operator", operator)
+
+
+# pylint: disable=broad-exception-raised
+class MergedConf:
+    def __init__(self, logger: Logger):
+        self.logger = logger
+        self.application = ApplicationConf("").parse_conf()
+        self.project = ProjectConf("").parse_conf()
+        self.operator = OperatorConf("").parse_conf()
+
+        self.merged_conf = {
+            "cli": {},
+            "application": self.merge_conf(self.application),
+            "project": self.merge_conf(self.project),
+            "operator": self.merge_conf(self.operator)
+        }
+
+        self.pre_final = self.create_final_conf()
+        self.final = self.validate_final_conf()
+
+    def validate_final_conf(self):
+        if validate_dict(self.pre_final, final_schema) is None:
+            return self.pre_final
+        raise Exception("Final config validation error.")
+
+    def create_final_conf(self):
+        final_conf = self.merge_conf(final.template)
+
+        for key, value in CONFIG_RULES.items():
+            rule = CONFIG_RULES.get(key)
+            if not rule:
+                raise Exception(f"No rules defined for conf key {key}, update CONFIG_RULES.")
+            mode = value.get("mode")
+            if not mode:
+                raise Exception(
+                    "Order is undefined. Rule mode order needs to be defined for rule mode.")
+            if not value.get(mode):
+                raise Exception("Mode is undefined")
+            final_conf[key] = self.get_final_value(key)
```

Comment:
`final_conf` values need to be mapped back onto `final.template` and then this `mapped_final_conf` needs to be validated with proper schema. Not sure if merged conf (final_conf) needs to be validated.

Reply:
Quick solution (feel free to implement your own): split final_conf key by "_" and iterate .get() on final.template until you get not-None return.

Reply:
If you are struggling with that check out 6f36b684504aa5e70cee970c8251af977b3fc4e4. Not gonna lie, it took me few hours, mainly because I'm stubborn and bad with debugger, so I did most debugging with `print`s in Linux terminal (bad idea). Anyway, code is rather ugly and not optimized but it does the job. If you decide to use it, please rename variables, functions, remove `print` statements etc.